aws/src/main/java/org/apache/iceberg/aws/glue/GlueCatalog.java

```diff
@@ -619,4 +635,21 @@ public void setConf(Configuration conf) {
   protected Map<String, String> properties() {
     return catalogProperties == null ? ImmutableMap.of() : catalogProperties;
   }
+
+  private Cache<TableOperations, FileIO> newFileIOCloser() {
+    return Caffeine.newBuilder()
+        .weakKeys()
+        .removalListener(
+            (RemovalListener<TableOperations, FileIO>)
+                (ops, fileIO, cause) -> {
+                  if (null != fileIO) {
+                    fileIO.close();
+                  }
+                })
+        .build();
+  }
+
+  private void trackFileIO(GlueTableOperations ops) {
```

Comment:
I don't think we need this wrapper method just for a put operation

Reply:
It's used at two places in the above code, otherwise I would have removed it.

Reply:
both places can just call `fileIOCloser.put(ops, ops.io());` rather than introducing this method. In `RESTSessionCatalog#trackFileIO(..)` the situation is slightly different, because `trackFileIO()` does some additional checking, which isn't necessary here, meaning that we don't need this method

Reply:
Makes senses. Removed the method.